frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -843,6 +843,29 @@ function ReaderHighlight:onHoldRelease()
     return true
 end
 
+function ReaderHighlight:onToggleHighlightAction()
+    local actions = {
+        highlight = "translate",
```

Comment:
Looks alright, just some renaming suggestions:
May be just rename that table to `next_actions` (plural), so you have below `next_action = next_actions[current_action]`
And the method `onCycleHighlightAction()` as we're not really toggling but rotating.